AutonomousDemo/SimulationLoop.cs

```diff
@@ -11,16 +11,23 @@ public class SimulationLoop
     {
         public Position lastPosition = new Position();
 
-        public void Run(Vehicle vehicle, double distance) { 
+        public void Run(Vehicle vehicle, double distance) {
 
-            while (true)
+            DeterminePath determinePath = new DeterminePath(distance);
+
+            vehicle.VehicleMotion.TravelDistance = distance;
+
+            while (vehicle.VehicleMotion.TravelDistance >= vehicle.VehicleMotion.DistanceTraveled)
```

Comment:
Why is "TravelDistance" a property of "VehicleMotion"? You should make a list of all of your classes and what they are actually for, and see if anything jumps out at you.